test/Sentry.Maui.Tests/SentryMauiScreenshotTests.cs

```diff
@@ -0,0 +1,87 @@
+using Microsoft.Extensions.Options;
+using Sentry.Internal.Http;
+
+namespace Sentry.Maui.Tests;
+
+public class SentryMauiScreenshotTests
+{
+    private class Fixture
+    {
+        public MauiAppBuilder Builder { get; }
+
+        public Fixture()
+        {
+            var builder = MauiApp.CreateBuilder();
+            builder.Services.AddSingleton(Substitute.For<IApplication>());
+
+            builder.Services.Configure<SentryMauiOptions>(options =>
+            {
+                options.Transport = new FakeTransport();
+                options.Dsn = ValidDsn;
+                options.AttachScreenshots = true;
+                options.AutoSessionTracking = false; //Get rid of session envelope for easier Assert
+                options.CacheDirectoryPath = null;   //Do not wrap our FakeTransport with a caching transport
+            });
+
+            Builder = builder;
+        }
+    }
+
+    private readonly Fixture _fixture = new();
+
+    [Fact]
+    public async Task CaptureException_WhenAttachScreenshots_ContainsScreenshotAttachmentAsync()
+    {
+        // Arrange
+        var builder = _fixture.Builder.UseSentry();
+
+        // Act
+        using var app = builder.Build();
+        var client = app.Services.GetRequiredService<ISentryClient>();
+        var sentryId = client.CaptureException(new Exception());
+        await client.FlushAsync();
+
+        var options = app.Services.GetRequiredService<IOptions<SentryMauiOptions>>().Value;
+        var transport = options.Transport as FakeTransport;
+
+        var envelope = transport.GetSentEnvelopes().FirstOrDefault(e => e.TryGetEventId() == sentryId);
+        envelope.Should().NotBeNull();
+
+        var envelopeItem = envelope!.Items.FirstOrDefault(item => item.TryGetType() == "attachment");
+
+        // Assert
+#if __MOBILE__
```

Comment:
else we don't assert anything, is this test supposed to only exist in mobile then? Looks lke we're missing something here for non mobile

Reply:
There is nothing to verify for mobile, I can wrap the whole test in #ifdef

Reply:
Yes please add the whole test then, otherwise it's running half a test and not asserting anything, I don't see the goal